res/controllers/Pioneer-DDJ-SB-scripts.js

```diff
@@ -898,88 +840,81 @@ PioneerDDJSB.rotarySelectorShiftedClick = function(channel, control, value, _sta
-        }
-    } else {
-        for (parameter = 0; parameter < 3; parameter++) {
-            if (PioneerDDJSB.fxButtonPressed[deck][parameter] && PioneerDDJSB.softTakeoverEmulation(deck, parameter, PioneerDDJSB.fxKnobMSB[deck])) {
-                engine.setParameter(
-                    '[EffectRack1_EffectUnit' + (deck + 1) + '_Effect1]',
-                    'parameter' + (parameter + 1),
-                    fullValue / 0x3FFF
-                );
+        components.Button.call(this);
+    };
+    this.EffectButton.prototype = new components.Button({
+        input: function(channel, control, value, status) {
+            if (this.isPress(channel, control, value, status)) {
+                this.isLongPressed = false;
+                this.longPressTimer = engine.beginTimer(this.longPressTimeout, function() {
+                    var effectGroup = "[EffectRack1_EffectUnit" + unitNumber + "_Effect" + this.buttonNumber + "]";
+                    script.toggleControl(effectGroup, "enabled");
+                    this.isLongPressed = true;
+                }, true);
+            } else {
+                if (!this.isLongPressed) {
+                    var focusedEffect = engine.getValue(eu.group, "focused_effect");
+                    if (focusedEffect === this.buttonNumber) {
+                        engine.setValue(eu.group, "focused_effect", 0);
+                    } else {
+                        engine.setValue(eu.group, "focused_effect", this.buttonNumber);
+                    }
+                }
+                this.isLongPressed = false;
+                engine.stopTimer(this.longPressTimer);
             }
+        },
+        outKey: "focused_effect",
+        output: function(value, _group, _control) {
+            this.send((value === this.buttonNumber) ? this.on : this.off);
+        },
+        sendShifted: true,
+        shiftControl: true,
+        shiftOffset: 28,
+    });
+
+    this.button = [];
+    for (var i = 1; i <= 3; i++) {
+        this.button[i] = new this.EffectButton(i);
+
+        var effectGroup = "[EffectRack1_EffectUnit" + unitNumber + "_Effect" + i + "]";
+        engine.softTakeover(effectGroup, "meta", true);
+        engine.softTakeover(eu.group, "mix", true);
+    }
+
+    this.knob = new components.Pot({
+        unshift: function() {
+            this.input = function(channel, control, value, _status) {
+                value = (this.MSB << 7) + value;
+
+                var focusedEffect = engine.getValue(eu.group, "focused_effect");
+                if (focusedEffect === 0) {
+                    engine.setParameter(eu.group, "mix", value / this.max);
+                } else {
+                    var effectGroup = "[EffectRack1_EffectUnit" + unitNumber + "_Effect" + focusedEffect + "]";
+                    engine.setParameter(effectGroup, "meta", value / this.max);
+                }
+            };
+        },
```

Comment:
there is also `inSetParameter` to simplify this a bit:
```suggestion
          inSetParameter: function(channel, control, value, _status) {
              if (engine.getValue(eu.group, "focused_effect") === 0) {
                  engine.setParameter(eu.group, "mix", value);
              } else {
                  var effectGroup = "[EffectRack1_EffectUnit" + unitNumber + "_Effect" + focusedEffect + "]";
                  engine.setParameter(effectGroup, "meta", value);
              }
          },
```
and if you don't overwrite anything in the `shift` method, you also don't need to touch the `unshift`.

Reply:
Solved in https://github.com/mixxxdj/mixxx/pull/12280/commits/00d56cce043849447b7d7620cffe125b4aea330d